### ISSUE TITLE
Errors in demo 

### ISSUE THREAD

**Issue by saglag** [flatironinstitute/CaImAn-MATLAB#104]
Please note that the developer of this package is on leave until January 2019 and might not be able to support you.

For better support, please use the template below to submit your issue. When your issue gets resolved please remember to close it.

- **Describe the issue that you are experiencing**
I'm trying to run through the demo script but it seems to encounter some errors that I can't figure out how it's being caused.

- **Copy error log below**

Error in montage/loadimages (line 1084)
M.im(nprev+i) = fn_structmerge(model,simi);

Error in montage (line 43)
loadimages(M,varargin{:})

Error in demo_script (line 90)
subplot(121); montage(extract_patch(A(:,keep),[d1,d2],[30,30]),'DisplayRange',[0,0.15]);

- **If you're not reporting an error, type your message below**

**Comment by epnev** [comment]
@saglag 
- Are you running the demo on the default dataset or your own data?
- Can you post the full error log?

**Comment by saglag** [comment]
Hi @epnev I ran it on the default. 

Here's the whole command output, it occurs right after the name request pops up. Image link:
![cZ8X6Nr7BO](https://user-images.githubusercontent.com/32366041/68040235-4856e080-fca4-11e9-8d2b-2fdc00780e52.png)

The following is the entire command window when the script starts on a fresh matlab instance:
>> demo_script
Starting parallel pool (parpool) using the 'local' profile ...
Connected to the parallel pool (number of workers: 4).
Warning: TIFF library warning - 'TIFFReadDirectory:  Unknown field with tag 50838 (0xc696) encountered.' 
> In Tiff (line 657)
  In read_file (line 28)
  In demo_script (line 10) 
Warning: TIFF library warning - 'TIFFReadDirectory:  Unknown field with tag 50839 (0xc697) encountered.' 
> In Tiff (line 657)
  In read_file (line 28)
  In demo_script (line 10) 
Warning: TIFF library warning - 'TIFFReadDirectory:  Unknown field with tag 50838 (0xc696) encountered.' 
> In Tiff/setDirectory (line 1298)
  In read_file (line 32)
  In demo_script (line 10) 
Warning: TIFF library warning - 'TIFFReadDirectory:  Unknown field with tag 50839 (0xc697) encountered.' 
> In Tiff/setDirectory (line 1298)
  In read_file (line 32)
  In demo_script (line 10) 
Estimating the noise power for each pixel from a simple PSD estimate...  done 
No spatial downsampling is performed. Consider spatial downsampling if the field of view is very large. 
No temporal downsampling is performed. Consider temporal downsampling if the recording is very long. 
starting resampling 
Initializing components with greedy method 
found 10 out of 40 neurons..
found 20 out of 40 neurons..
found 30 out of 40 neurons..
found 40 out of 40 neurons..
Refining initial estimates with HALS...Warning: File: determine_search_location.m Line: 92 Column: 20
The temporary variable 'ind_t' will be cleared at the beginning of each iteration of the parfor-loop. If 'ind_t' is used before it is set, a runtime error will occur. For more
information, see Parallel for Loops in MATLAB, "Uninitialized Temporaries". 
> In HALS (line 33)
  In initialize_components (line 141)
  In demo_script (line 39) 
  done 
Updated spatial components 
Warning: File: update_temporal_components.m Line: 234 Column: 40
The temporary variable 'model_ar' will be cleared at the beginning of each iteration of the parfor-loop. If 'model_ar' is used before it is set, a runtime error will occur. For
more information, see Parallel for Loops in MATLAB, "Uninitialized Temporaries". 
> In demo_script (line 62) 
21 out of 40 components updated 
37 out of 40 components updated 
40 out of 40 components updated 
21 out of 40 components updated 
37 out of 40 components updated 
40 out of 40 components updated 
21 out of 40 components updated 
37 out of 40 components updated 
40 out of 40 components updated 
21 out of 40 components updated 
37 out of 40 components updated 
40 out of 40 components updated 
Warning: cannot fit the buttons inside the container 
> In fn_control (line 785)
  In montage/init_control (line 115)
  In montage>@(u,e)M.init_control (line 57)
  In interface/interface_end (line 105)
  In montage (line 34)
  In demo_script (line 90) 
Subscripted assignment between dissimilar structures.

Error in montage/loadimages (line 1084)
                M.im(nprev+i) = fn_structmerge(model,simi);

Error in montage (line 43)
                    loadimages(M,varargin{:})

Error in demo_script (line 90)
    subplot(121); montage(extract_patch(A(:,keep),[d1,d2],[30,30]),'DisplayRange',[0,0.15]);
 
>> Let me know if you need more info!

**Comment by epnev** [comment]
Are you trying to call this through some interface you're building?

**Comment by saglag** [comment]
Nope, I loaded the package through the add-on manager. I opened the demo_script.m and clicked run

**Comment by epnev** [comment]
I'm not sure how the add-on manager works. It's likely you don't get the latest version of the code that way. It's probably best if you download directly from github. Also, if you're a python user I'd recommend using the python version instead.

**Comment by saglag** [comment]
Deleted the folder/files, downloaded the github page one, added to path and still get the same issue. Unfortunately, I only have limited experience working in Spyder for python and struggle at implementing even simple scripts.

**Comment by epnev** [comment]
Apologies for the delayed response. Something is wrong with the `montage` function but I cannot tell what from the log you're providing. Try to see if you can use the function on its own, outside of caiman. The demos should run fine without it; it's just a figure that is being plotted.
